prow/pod-utils/gcs/upload_test.go

```diff
@@ -95,8 +114,16 @@ func TestUploadToGcs(t *testing.T) {
 			t.Errorf("%s: expected an error but got none", testCase.name)
 		}
 
-		if count != (testCase.passingTargets + testCase.failingTargets) {
-			t.Errorf("%s: had %d passing and %d failing targets but only ran %d targets, not %d", testCase.name, testCase.passingTargets, testCase.failingTargets, count, testCase.passingTargets+testCase.failingTargets)
+		finalPassingCount := testCase.passingTargets
+		finalFailingCount := testCase.failingTargets * uploadRetries
+
+		// flaky test will fail until final try so account for those extra attempts
+		if testCase.flakyTargets > 0 {
+			finalPassingCount = testCase.passingTargets + testCase.flakyTargets + (uploadRetries - 1)
```

Comment:
If each flaky target succeeds on the last try, why isn't `finalPassingCount` just `testCase.passingTargets + testCase.flakyTargets` ?

Similarly, isn't
```go
finalFailingCount := testCase.failingTargets * uploadRetries + testCase.flakyTargets * (uploadRetries - 1)
```

Reply:
for every flaky target, count will increment to be == (uploadRetries - 1) which in this case is '2' for every flakyTarget. and
for failingTargets count ends up as '3' for every one.

Reply:
Right, so uh why is the multiplication I posted incorrect and why is the addition in the code correct?

Reply:
I think I get what you are asking, and maybe the confusion is just that we are updating
a generic "count" variable when the upload func is called. In the case of the fail() or success()
funcs it's calling update() which increments count. For the fail case, count ends up euqal
to uploadRetries. For the passing case, it just increments once. In the flaky case, we end up
only incrementing count 1 less than uploadRetries and that's the check we make in the end.

Reply:
I'm still lost, so why isn't the check

```go
if count != (testCase.passingTargets + (testCase.failingTargets + testCase.flakyTargets) * uploadRetries) {
```

However, that doesn't really end up testing much for this PR, should record passes and fails separately, and we expect

```go
if passingCount != (testCase.passingTargets + testCase.flakyTargets) {
    /* whatever */
}

if failingCount != (testCase.failingTargets * uploadRetries + testCase.flakyTargets * (uploadRetries - 1)) {
    /* whatever */
}
```

Reply:
I think the code you wrote in this PR only works if there is one flaky test -- try bumping the number of flakes up

Reply:
thanks for catching this. the retry logic was working as expected, but the test code was wrong. latest patch set fixes the tests.